core/src/main/java/net/adoptopenjdk/icedteaweb/jvm/JvmUtils.java

```diff
@@ -41,29 +39,24 @@
      * @throws IllegalArgumentException if the VM arguments are invalid or dangerous
      */
     public static void checkVMArgs(final String vmArgs) throws IllegalArgumentException {
-        final boolean isJava9orGreater = JAVA_9_OR_GREATER.contains(JVM_VERSION);
-        checkVMArgs(vmArgs, isJava9orGreater);
-    }
-
-    // visible for testing
-    static void checkVMArgs(final String vmArgs, final boolean isJava9orGreater) {
```

Comment:
I know that the method already was designed like this but maybe it makes sense to provide a return value instead of just throwing an exception.

What I would like:

`static Collection<VmArgumentViolation> checkVMArgs(final String vmArgs)`

The returned collection is never null and contains violations for each arg that is not allowed. `VmArgumentViolation` contains the arg name (and maybe the index).

Reply:
it is called from Parser. didn't want to touch it. besides the logic is that if one arg is bad all args are ignored.